### PR TITLE
Add folder tekton for automated mirror jobs

### PR DESCRIPTION
**Pull request by arilivigni**
- tekton_v0_3 for image information needed for mirroring
- Owners file

### CODE REVIEW

File: cluster/ci/config/mirroring/tekton/tekton_v0_3 [hunk 1]
```diff
@@ -0,0 +1,8 @@
+registry.svc.ci.openshift.org/openshift/knative:tektoncd-pipeline-bash:v0.3.1 quay.io/openshift-pipelines/tektoncd-pipeline-bash:v0.3.1
```
Comment (vdemeester) [inline cluster/ci/config/mirroring/tekton/tekton_v0_3]: the CI images are currently weirder : `registry.svc.ci.openshift.org/openshift/knative:tektoncd-pipeline-bash` (no "version"). 

https://github.com/openshift/release/pull/3609 is fixing that, but also changing the prefix name, so in the future, images will be `registry.svc.ci.openshift.org/openshift/tekton-v0.3.1:tektoncd-pipeline-bash` instead

File: cluster/ci/jobs/image-mirror.yaml [hunk 2]
```diff
@@ -281,3 +281,50 @@ items:
...
+          metadata:
+            name: "image-mirror-tekton-v0.3-quay"
+            labels:
+              app: "image-mirror-tekton-v0.3-quay"
+          spec:
+            restartPolicy: OnFailure
+            volumes:
+              - name: push
+                secret:
+                  secretName: registry-push-credentials-quay.io-openshift-pipeline
```
Comment (vdemeester) [inline cluster/ci/jobs/image-mirror.yaml]: Is this secret present ? (I think I didn't yet created a robot account on quay :sweat_smile:)

File: cluster/ci/config/mirroring/tekton/tekton_v0_3 (hunk 1 above)
Comment (arilivigni) [reply cluster/ci/config/mirroring/tekton/tekton_v0_3]: Updated

File: cluster/ci/jobs/image-mirror.yaml (hunk 2 above)
Comment (arilivigni) [reply cluster/ci/jobs/image-mirror.yaml]: I will look into it

**Comment by petr-muller** [comment]
@arilivigni I don't have a great insight to how mirroring is set up. The CronJob looks fine, but I think you'll hit the same problem like knative folks hit where the mirror config file would not be present in the mounted config map (see https://github.com/openshift/release/issues/3834#issuecomment-494730652) 

@stevekuznetsov suggested in that issue that these mirroring artifacts should also be placed in a separate namespace so you can own and debug the config.

**Comment by matzew** [comment]
@petr-muller Thanks for your help on Knative! I did transform the knowledge and did help on this PR.

things should be identically now :-)

**Comment by arilivigni** [comment]
@vdemeester I can add a commit for v0.4.0 is you want since it hasn't been merged yet

**Comment by vdemeester** [comment]
@arilivigni yeah :angel:

**Comment by arilivigni** [comment]
@vdemeester added v0.4 as well

File: cluster/ci/jobs/image-mirror.yaml [hunk 3]
```diff
@@ -234,3 +234,144 @@ items:
               - mirror
               - -f=/etc/imagemirror/origin_4_2
               - --skip-multiple-scopes # quay chokes when requesting multiple
+- apiVersion: batch/v1beta1
```
Comment (petr-muller) [inline cluster/ci/jobs/image-mirror.yaml]: Why is this PR adding some knative jobs? Is it intentional?

Comment (vdemeester) [reply cluster/ci/jobs/image-mirror.yaml]: Good point, I don't think it should add anything in there, especially related to knative…

Comment (matzew) [reply cluster/ci/jobs/image-mirror.yaml]: 100% agree

I did actually factor the tekton bits... out ->   https://github.com/arilivigni/release/pull/1/files

Comment (arilivigni) [reply cluster/ci/jobs/image-mirror.yaml]: I will remove those apologies

**Comment by arilivigni** [comment]
@petr-muller @vdemeester  I removed knative references in the image-mirror job

**Comment by arilivigni** [comment]
@smarterclayton @petr-muller is it possible to get this merged before the Makefile differs again I believe this meets all the requirements AFAIK

File: cluster/ci/jobs/tekton-image-mirror.yaml [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+            volumes:
+              - name: push
+                secret:
+                  secretName: registry-push-credentials-quay.io-openshift-pipeline
+              - name: config
+                configMap:
+                  name: tekton-image-mirror
+            containers:
+              - name: "mirror"
+                image: "openshift/origin-cli:v4.0"
```
Comment (smarterclayton) [inline cluster/ci/jobs/tekton-image-mirror.yaml]: This should be v4.1, v4.0 is super out of date

**Comment by smarterclayton** [comment]
One comment (using the wrong old image)

**Comment by arilivigni** [comment]
@smarterclayton I updated to 4.1 for the cli thanks!

**Pull request merged**